WooCommerce/woocommerce_rms_normal/class-gateway.php

```diff
@@ -376,13 +434,156 @@ function check_molpay_response_callback() {
           $status = $_POST['status'];
           if ( !$verifyresult )
               $status = "-1";
-          
-          $WCOrderId = $this->get_WCOrderIdByOrderId($_POST['orderid']);
+
+		    $extraP = isset($_POST['extraP']) ? $_POST['extraP'] : '';
+
+			if (isset($_POST['extraP']) && !empty($_POST['extraP']) && ($this->recurring == 'yes')) {
```

Comment:
rearrange the code, make it pretty

Reply:
Make it pretty : 639d7d75996fc23dcc2508b73c73328c707bf713